Detectors/MUON/MCH/PreClustering/CMakeLists.txt

```diff
@@ -9,14 +9,14 @@
 # submit itself to any jurisdiction.
 
 o2_add_library(MCHPreClustering
-               SOURCES src/PreClusterFinder.cxx
-                       src/PreClusterFinderMapping.cxx
-               PUBLIC_LINK_LIBRARIES O2::MCHMappingFactory O2::MCHBase O2::Framework)
+        SOURCES src/PreClusterFinder.cxx
+        src/PreClusterFinderMapping.cxx
+        PUBLIC_LINK_LIBRARIES O2::MCHMappingImpl3 O2::MCHBase O2::Framework)
```

Comment:
Does this mean that we will need two separate MCHPreClustering libraries, one for Run1-2 data and one for Run3?

The pre-clustering takes digits as input, which in turn provide the hits in (deID,padID) coordinates. Hence the differences in the channel mapping between Run1-2 and Run3 are already "embedded" into the digit, and the pre-clustering should be independent of the actual channel mapping.

Am I missing something?

Reply:
As it is now, yes, it means you need two libraries (or two executables) linked with the two  different mapping implementations. Here the idea is to ensure that the Run3 code is not paying for anything Run2 related. 
If that's really an issue we could imagine (but for testing purposes, not as baseline solution) to have a piece of code that dynamically loads the correct library at runtime, but I did not go that route to keep things simple (and explicitly avoid to mix up Run2 and Run3 stuff)

Reply:
Could you already introduce the two MCHPreClustering libraries in your PR?

Another possibility would be to separate the two steps in the mapping, that is:
1. the part that allows to go from `(boardID, channelID)` to `padID`
2. and the one that provides the pad coordinates, that is from `padID` to `(padX, padY)`

Only step 1. depends on the actual FEC channel mapping, while step 2. is common for both. AFAIK the PreClustering would only require step 2., right? Then it could be made independent of the FEC type.

However, this is probably a lot of work for something that is expected to be removed in the near future...

Reply:
The preclustering is supposed to work with run3 data. The conversion run2-run3 is done is DigitSamplerSpec and PreClusterSinkSpec. So I suppose the preclustering should link to impl4 and the spec to impl3. Is it possible?

Well in fact the preclustering does not use the pad channel number but only the PadID (index) so it guess it's more on the principle but it does not really matter the implementation it is linked with?

I will run some tests on my side too.

Reply:
I'm afraid it's not, by construction, possible to link to both implementations. They are mutually exclusive, as they implement the same functions...